app/src/main/java/com/github/sdpsharelook/revision/RevisionQuizFragment.kt

```diff
@@ -0,0 +1,67 @@
+package com.github.sdpsharelook.revision
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.lifecycleScope
+import com.github.sdpsharelook.R
+import com.google.android.material.floatingactionbutton.FloatingActionButton
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.launch
+
+@AndroidEntryPoint
+class RevisionQuizFragment : RevisionQuizFragmentLift()
+
+open class RevisionQuizFragmentLift : Fragment() {
+    private val viewModel: RevisionQuizViewModel by viewModels()
+    private var showHelp = false
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        view.findViewById<FloatingActionButton>(R.id.helpToggleButton)
+            .setOnClickListener {
+                handleHelpToggle(view)
+            }
+        view.findViewById<ConstraintLayout>(R.id.quizLayout)
+            .setOnClickListener {
+                view.isClickable = false
+                view.isFocusable = false
+                viewModel.onEvent(QuizEvent.Continue)
+            }
+        lifecycleScope.launch {
+            viewModel.uiEvent.apply { TODO() }
+        }
+    }
+
+    private fun handleHelpToggle(view: View) {
+        if (showHelp) {
+            showHelp = false
+            view.findViewById<Button>(R.id.answerQualityButton0).setText(R.string.quality0)
```

Comment:
You could maybe put all the ids on a list.

Reply:
That is clever

Reply:
Done